apps/yapms/src/routes/app/[country]/[map]/[year]/initialize/LoadRegions.ts

```diff
@@ -114,6 +115,12 @@ function loadRegions(node: HTMLDivElement): void {
 		console.error('Error Parsing Candidate Data from Map:\n\n' + error);
 	}
 
+	//If map being loaded has the default-mode property set, change the mode.
+	const defaultMode = node.querySelector('svg')?.getAttribute('default-mode');
+	if (defaultMode !== null) {
+		ModeStore.set(defaultMode as Mode);
```

Comment:
Here, you need to check for the undefined condition, and each possible string value of Mode. Currently the cast to Mode is causing undefined type to be hidden.

```
if (defaultMode !== null && defaultMode !== undefined && (defaultMode === "edit" ... ))
```
This will reduce the string type down to the Mode type.

AFAIK this is the only way... unless there is some fancy typescripting that can be done with the Mode type.

Reply:
Ah rats.. will check for a way to maybe do this in zod tomorrow.

Reply:
Oh yeah, I forgot about zod lol. Yeah, that's a better idea.